### ISSUE TITLE
Typing question

### ISSUE THREAD

**Issue by akomm** [relay-tools/relay-hooks#41]
I assume typing is currently quite WIP or is there a reason behind certain decisions, like `useRefetch` being totally any-typed or `useQuery`'s retry requiring an object config which should be optional, or `useFragment` not taking a generic type, like `useQuery`, but instead you have to annotate the return type?

I have following "wrapping" with API change and some more typing for `useRefetch`, note its just written quickly in ~1-2 mins and so bear that in mind, its sure not perfect ^^.

```ts
import {useCallback} from "react";
import {useRefetch} from "relay-hooks/lib";
import {RefetchOptions} from "relay-hooks/lib/useOssFragment";
import {OperationType} from "relay-runtime";

type RefetchFunction = <TOperationType extends OperationType>(refetchArgs: {
  query: any,
  refetchVariables: TOperationType['variables'],
  renderVariables?: TOperationType['variables'],
  observableOrCallback?: any,
  options: RefetchOptions,
}) => {dispose: () => void};

export default function<TFragmentResult = any>(fragmentDef: any, fragmentRef: any): [TFragmentResult, RefetchFunction] {
  const [result, _refetch] = useRefetch(fragmentDef, fragmentRef);
  const refetch = useCallback<RefetchFunction>(args => {
    return _refetch(
      args.query,
      args.refetchVariables,
      args.renderVariables,
      args.observableOrCallback,
      args.options,
    );
  }, [_refetch]);
  return [result, refetch];
}
```

Especially since many things are defined in the typing as non-optional, even if they are, working with non-object arguments is pain in butt, this is why I changed it for me. But you might have reasons why it is like it is if its not just due to WIP?

**Comment by morrys** [comment]
Hi @akomm,
there are a lot of things to do and I can't always find the time to correctly define all the typings :).
So the contributions are welcome :)

You can propose a PR so it's easier to evaluate its integration

**Comment by akomm** [comment]
Okay. But then I need to know if the multi-arg API vs. object-arg API is intended or not?

**Comment by morrys** [comment]
This library tries to follow as much as possible what is realized in react-relay (unless of the promises, being experimental).

Their direction is this: https://github.com/facebook/relay/blob/master/packages/relay-experimental/useFragment.js#L54 (multi-args)

**Comment by morrys** [comment]
hi @akomm,
today I released version 2.1.0 where I solved this issue:
https://github.com/relay-tools/relay-hooks/releases/tag/v2.1.0

For any problem, open a new issue.
thanks
